pageserver/src/bin/pageserver.rs

```diff
@@ -449,6 +459,21 @@ fn start_pageserver(conf: &'static PageServerConf) -> Result<()> {
     // Initialize logger
     let (_scope_guard, log_file) = logging::init(LOG_FILE_NAME, conf.daemonize)?;
 
+    let term_now = Arc::new(AtomicBool::new(false));
+    for sig in TERM_SIGNALS {
+        // When terminated by a second term signal, exit with exit code 1.
+        // This will do nothing the first time (because term_now is false).
+        flag::register_conditional_shutdown(*sig, 1, Arc::clone(&term_now))?;
+        // But this will "arm" the above for the second time, by setting it to true.
+        // The order of registering these is important, if you put this one first, it will
+        // first arm and then terminate ‒ all in the first round.
+        flag::register(*sig, Arc::clone(&term_now))?;
+    }
+
+    let mut sigs = vec![];
+    sigs.extend(TERM_SIGNALS);
+    let mut signals = SignalsInfo::<WithOrigin>::new(&sigs)?;
```

Comment:
Small style nit: looks like the signals processing could be reduced to 
`for info in SignalsInfo::<WithOrigin>::new(TERM_SIGNALS)?.into_iter() {` lower, this way we don't allocate anything and don't have a `mut` thing sticking around.